code/modules/mob/vision_cone.dm

```diff
@@ -0,0 +1,91 @@
+/mob
+	var/obj/screen/fov = null//The screen object because I can't figure out how the hell TG does their screen objects so I'm just using legacy code.
+
+client/
+	var/list/hidden_atoms = list()
+	var/list/hidden_mobs = list()
+	var/list/hidden_images = list()
+
+
+
+//Procs
+/atom/proc/InCone(atom/center = usr, dir = NORTH)
+	if(get_dist(center, src) == 0 || src == center) return 0
+	var/d = get_dir(center, src)
+	if(!d || d == dir) return 1
+	if(dir & (dir-1))
+		return (d & ~dir) ? 0 : 1
+	if(!(d & dir)) return 0
+	var/dx = abs(x - center.x)
+	var/dy = abs(y - center.y)
+	if(dx == dy) return 1
+	if(dy > dx)
+		return (dir & (NORTH|SOUTH)) ? 1 : 0
+	return (dir & (EAST|WEST)) ? 1 : 0
+
+/mob/dead/InCone(mob/center = usr, dir = NORTH)//So ghosts aren't calculated.
+	return
+
+/proc/cone(atom/center = usr, dir = NORTH, list/list = oview(center))
+	for(var/atom/A in list)
+		if(!A.InCone(center, dir))
+			list -= A
+	return list
+
+/mob/proc/update_vision_cone()
+	return
+
+/mob/living/update_vision_cone()
+	if(src.client)
+		var/image/I = null
+		for(I in src.client.hidden_atoms)
+			I.override = 0
+			client.images -= I
```

Comment:
var/list/imagecopy = client.images.Copy()

then do all operations on the copy of the list, then assign client.images to the copy.

this saves the overhead of byond parsing the image list so it does it only once.